File: arkitekt_core/src/main/java/com/eltonkola/arkitekt/ScreenNavigation.java
package com.eltonkola.arkitekt;

import android.view.MenuInflater;

/**
 * Created by elton on 9/29/17.
 */

public interface ScreenNavigation {

    void close();

    void goTo(final String path, final Object param);

    MenuInflater getMenuInflater();

}

File: arkitekt_core/src/main/java/com/eltonkola/arkitekt/ScreenNavigation.kt
package com.eltonkola.arkitekt

import android.support.v7.widget.Toolbar
import android.view.MenuInflater
import com.eltonkola.arkitekt.views.AppScreenContainer

interface ScreenNavigation {

    val menuInflater: MenuInflater

    val isInPortraitMode: Boolean

    fun close()

    fun goTo(path: String, param: Any?)

    fun goToAndClose(path: String, param: Any?)

    fun toastShort(msg: String)

    fun toastLong(msg: String)

    fun showScreen(parent: AppScreen<*>, app_screen_container: AppScreenContainer, appScreenPath: String, param: Any?)

    fun setActionBar(toolbar: Toolbar)
}
